### PR TITLE
Added evaluation folder and core files

### PR DESCRIPTION
**Pull request by BreckoD**
Added 'eval' folder with 'eval_runner.py' script and 'analyst_eval.ipynb' main file.

### CODE REVIEW

File: src/sttn/eval/eval_runner.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
+import nbformat
```
Comment (yuribogomolov) [inline src/sttn/eval/eval_runner.py:1]: Does your notebook consume this script? We probably don't need to check it in now if it's not working.

File: src/sttn/eval/analyst_eval.ipynb [hunk 2]
```diff
@@ -0,0 +1,576 @@
+{
```
Comment (yuribogomolov) [inline src/sttn/eval/analyst_eval.ipynb:1]: Could you please move the logic to Python modules & classes?
Notebooks are not the best interface to manage Python code. Ideally, the runner notebook should contain only a few lines of code to start the evaluation.

File: src/sttn/eval/analyst_eval.ipynb [hunk 3]
```diff
@@ -0,0 +1,576 @@
...
+    "### Upload `langsmith dataset`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test_dataset = client.read_dataset(dataset_name=dataset_name)\n",
```
Comment (yuribogomolov) [inline src/sttn/eval/analyst_eval.ipynb:63]: Where are the datasets stored? Can we include them in this PR?

Comment (BreckoD) [reply src/sttn/eval/analyst_eval.ipynb:63]: They are stored in LangSmith

File: src/sttn/eval/analyst_eval.ipynb (hunk 2 above)
Comment (BreckoD) [reply src/sttn/eval/analyst_eval.ipynb:1]: Done! The new version has all evaluators in the separate file

File: src/sttn/eval/eval_runner.py (hunk 1 above)
Comment (BreckoD) [reply src/sttn/eval/eval_runner.py:1]: Yes, the runner script needs this library to run the notebook

**Pull request merged**